src/MainWindow.vala

```diff
@@ -132,8 +120,36 @@ public class AppCenter.MainWindow : Gtk.ApplicationWindow {
     }
 
     construct {
-        icon_name = "system-software-install";
+        settings = new GLib.Settings ("io.elementary.appcenter.settings");
+
+        int window_x, window_y;
+        int window_width, window_height;
+        settings.get ("window-position", "(ii)", out window_x, out window_y);
+        settings.get ("window-size", "(ii)", out window_width, out window_height);
+
         set_size_request (910, 640);
+
+        if (window_x != -1 ||  window_y != -1) {
+            move (window_x, window_y);
+        }
+
+        resize (window_width, window_height);
+
+        show.connect (() => {
```

Comment:
I don't think this is necessary. In other apps we're overriding the configure event method as in https://github.com/elementary/appcenter/pull/904/files#diff-4089e5f35cbce620c0c1758fb579b361R222